src/main.rs

```diff
@@ -315,7 +338,8 @@ fn switch_off_leds() {
 
 fn check_user_presence(cid: ChannelID) -> Result<(), Ctap2StatusCode> {
     // The timeout is N times the keepalive delay.
-    const TIMEOUT_ITERATIONS: usize = ctap::TOUCH_TIMEOUT_MS as usize / KEEPALIVE_DELAY_MS as usize;
+    const TIMEOUT_ITERATIONS: usize =
+        ctap2::ctap::TOUCH_TIMEOUT_MS as usize / KEEPALIVE_DELAY_MS as usize;
```

Comment:
This `ctap2` is the name from the `Cargo.toml` right?

I was wondering how confusing all the `ctap`s will be. There is the
- `ctap`: mod
- `Ctap`: new struct
- `ctap2`: lib name
- `ctap1` which is almost the same, and actually more on the level of CtapState than its child, really

And technically, the names are wrong. The lib should actually be `ctap` with a `Ctap` struct, containing `ctap1` and `ctap2` next to each other. Then there is also no `ctap2::ctap::ctap1` anymore.

I'll open extra comments for HID and CTAP1, but feel free to respond only at one place since it's all the same question really. Also not necessarily for this PR, but I think it's good to discuss goal and big picture here.

Reply:
Yes this is definitely not for this PR and should be addressed at some point. When the crate was split between main and lib for fuzzing it was done half-way. In particular the modules were used by both main and lib. This PR fixes this. As such, when main is referring to the library, it has to use the crate name ctap2. I agree that in the end we probably want the library to be called opensk or opensk-ctap or something like that, and it would contain just what the user needs to know with the minimum amount of modules. I'm not even sure ctap2 and ctap1 should be visible to the user. Probably only as cargo features. But that's to be designed later.

Reply:
Sounds good, leaving this open for reference.